### ISSUE TITLE
`UnicodeDecodeError.object` inferred as `str` instead of `bytes`

### ISSUE THREAD

**Issue by JulianJvn** [pylint-dev/pylint#9342]
### Bug description

In #3639, it was reported that accessing `UnicodeDecodeError.object` resulted in an error. pylint-dev/astroid#797 fixed this, unfortunately incorrectly. Pylint thinks `UnicodeDecodeError.object` is of type `str`, when it is actually of type `bytes`. (A `UnicodeDecodeError` occurs when decoding a `bytes` object; `str` is the result.)

Example code (file `test.py`):

```python
"""Shows that Pylint incorrectly deduces the type of
``UnicodeDecodeError.object`` as ``str`` instead of ``bytes``."""

try:
    print(b"\x80".decode())
except UnicodeDecodeError as exception:
    print("Error: " + exception.object.hex())
```

Output when run via `python test.py`: `Error: 80`

I don't know anything about astroid, but I'd guess the fix is to change [this line](https://github.com/pylint-dev/astroid/blob/31ba1dbd8ae86b55330be861e37d6bc372d9a8a5/astroid/interpreter/objectmodel.py#L821C18-L821C18) as follows:

```diff
-return node_classes.Const("")
+return node_classes.Const(b"")
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:7:22: E1101: Instance of 'str' has no 'hex' member (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

No Pylint errors.

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.2
Python 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)]
```

### OS / Environment

Windows 11, Git Bash
